packages/preview-middleware/src/base/flp.ts

```diff
@@ -344,11 +344,24 @@ export class FlpSandbox {
      */
     private addRoutesForAdditionalApps() {
         for (const app of this.config.apps) {
+            let manifest: Manifest | undefined;
             if (app.local) {
-                const manifest = JSON.parse(readFileSync(join(app.local, 'webapp/manifest.json'), 'utf-8'));
-                this.addApp(manifest, app);
+                manifest = JSON.parse(readFileSync(join(app.local, 'webapp/manifest.json'), 'utf-8'));
                 this.router.use(app.target, serveStatic(join(app.local, 'webapp')));
                 this.logger.info(`Serving additional application at ${app.target} from ${app.local}`);
+            } else if (app.componentId) {
+                manifest = {
+                    'sap.app': {
+                        id: app.componentId,
+                        title: app.intent ? `${app.intent.object}-${app.intent.action}` : app.componentId
+                    }
+                } as Manifest;
+            }
+            if (manifest) {
+                this.addApp(manifest, app);
+                this.logger.info(`Adding additional intent: ${app.intent?.object}-${app.intent?.action}`);
+            } else {
+                this.logger.info(`Invalid application config: ${app}`);
```

Comment:
I got this output:
```
info preview-middleware Invalid application config: [object Object]
```
The log could be a bit more helpful at indicating which app is causing issues. Maybe also explicitly stating that `local` or `componentId` must be defined.

Reply:
Fixed with https://github.com/SAP/open-ux-tools/pull/1764/commits/e58caa2a94a017a6792b48087a6fd71f994f1712.